src/key.js

```diff
@@ -623,6 +623,86 @@ Key.prototype.revoke = function() {
 };
 
 /**
+ * Signs the public key
+ * @param  {Array<module:key~Key>} privateKey decrypted private keys for signing
+ * @return {module:key~Key} new public key with new certificate signature
+ */
+Key.prototype.sign = function(privateKeys) {
+  var primaryUser, primaryKey, dataToSign, signingKeyPacket, signaturePacket, user, key;
+  if (this.isPrivate()) {
+    throw new Error('Only public key can be signed');
+  }
+  primaryUser = this.getPrimaryUser();
+  if (!primaryUser) {
+    throw new Error('Could not find primary user');
+  }
+  primaryKey = this.primaryKey;
+  dataToSign = {};
+  dataToSign.userid = primaryUser.user.userId || primaryUser.user.userAttribute;
+  dataToSign.key = primaryKey;
```

Comment:
> Key.prototype.signAllUsers is well-looking option to set some day another user as primary and not collect again all trusted signatures.

Yes, that is one option to create a new signature for all userIds. But it should also be enough to sign only one userId with the primary flag set. In the end, the newest signature with the primary flag sets the useId as "primary".

> About functionality located at User.prototype, we will have to also always pass primaryKey (of public key) to prototype methods, cause they need both to sign and verify.

Unfortunately yes.

> If I understand correctly userIds cant be signed without related keyid, otherwise someone could put own primaryKey and self-signature to trusted public key.

What do you mean by "related keyid"? The keyid of the primary key? I think two primaryKey packets would be a violation of the spec. If you replace the primaryKey and add a valid userId+signature, then you basically have a new key. But if available, subkey binding signatures have to match as well.